{
	"document": {
		"aggregate_severity": {
			"namespace": "https://www.suse.com/support/security/rating/",
			"text": "moderate"
		},
		"category": "csaf_security_advisory",
		"csaf_version": "2.0",
		"distribution": {
			"text": "Copyright 2024 SUSE LLC. All rights reserved.",
			"tlp": {
				"label": "WHITE",
				"url": "https://www.first.org/tlp/"
			}
		},
		"lang": "en",
		"notes": [
			{
				"category": "summary",
				"text": "Security update for python-urllib3",
				"title": "Title of the patch"
			},
			{
				"category": "description",
				"text": "This update for python-urllib3 fixes the following issues:\n\n- CVE-2024-37891: Fixed proxy-authorization request header not stripped during cross-origin redirects (bsc#1226469).\n",
				"title": "Description of the patch"
			},
			{
				"category": "details",
				"text": "SUSE-2024-2320,SUSE-SLE-Module-Public-Cloud-15-SP4-2024-2320,SUSE-SLE-Module-Python3-15-SP5-2024-2320,SUSE-SLE-Module-Python3-15-SP6-2024-2320,openSUSE-SLE-15.5-2024-2320,openSUSE-SLE-15.6-2024-2320",
				"title": "Patchnames"
			},
			{
				"category": "legal_disclaimer",
				"text": "CSAF 2.0 data is provided by SUSE under the Creative Commons License 4.0 with Attribution (CC-BY-4.0).",
				"title": "Terms of use"
			}
		],
		"publisher": {
			"category": "vendor",
			"contact_details": "https://www.suse.com/support/security/contact/",
			"name": "SUSE Product Security Team",
			"namespace": "https://www.suse.com/"
		},
		"references": [
			{
				"category": "external",
				"summary": "SUSE ratings",
				"url": "https://www.suse.com/support/security/rating/"
			},
			{
				"category": "self",
				"summary": "URL of this CSAF notice",
				"url": "https://ftp.suse.com/pub/projects/security/csaf/suse-su-2024_2320-1.json"
			},
			{
				"category": "self",
				"summary": "URL for SUSE-SU-2024:2320-1",
				"url": "https://www.suse.com/support/update/announcement/2024/suse-su-20242320-1/"
			},
			{
				"category": "self",
				"summary": "E-Mail link for SUSE-SU-2024:2320-1",
				"url": "https://lists.suse.com/pipermail/sle-security-updates/2024-July/018878.html"
			},
			{
				"category": "self",
				"summary": "SUSE Bug 1226469",
				"url": "https://bugzilla.suse.com/1226469"
			},
			{
				"category": "self",
				"summary": "SUSE CVE CVE-2024-37891 page",
				"url": "https://www.suse.com/security/cve/CVE-2024-37891/"
			}
		],
		"title": "Security update for python-urllib3",
		"tracking": {
			"current_release_date": "2024-07-08T11:14:03Z",
			"generator": {
				"date": "2024-07-08T11:14:03Z",
				"engine": {
					"name": "cve-database.git:bin/generate-csaf.pl",
					"version": "1"
				}
			},
			"id": "SUSE-SU-2024:2320-1",
			"initial_release_date": "2024-07-08T11:14:03Z",
			"revision_history": [
				{
					"date": "2024-07-08T11:14:03Z",
					"number": "1",
					"summary": "Current version"
				}
			],
			"status": "final",
			"version": "1"
		}
	},
	"product_tree": {
		"branches": [
			{
				"branches": [
					{
						"branches": [
							{
								"category": "product_version",
								"name": "python311-urllib3-2.0.7-150400.7.18.1.noarch",
								"product": {
									"name": "python311-urllib3-2.0.7-150400.7.18.1.noarch",
									"product_id": "python311-urllib3-2.0.7-150400.7.18.1.noarch"
								}
							}
						],
						"category": "architecture",
						"name": "noarch"
					},
					{
						"branches": [
							{
								"category": "product_name",
								"name": "SUSE Linux Enterprise Module for Public Cloud 15 SP4",
								"product": {
									"name": "SUSE Linux Enterprise Module for Public Cloud 15 SP4",
									"product_id": "SUSE Linux Enterprise Module for Public Cloud 15 SP4",
									"product_identification_helper": {
										"cpe": "cpe:/o:suse:sle-module-public-cloud:15:sp4"
									}
								}
							},
							{
								"category": "product_name",
								"name": "SUSE Linux Enterprise Module for Python 3 15 SP5",
								"product": {
									"name": "SUSE Linux Enterprise Module for Python 3 15 SP5",
									"product_id": "SUSE Linux Enterprise Module for Python 3 15 SP5",
									"product_identification_helper": {
										"cpe": "cpe:/o:suse:sle-module-python3:15:sp5"
									}
								}
							},
							{
								"category": "product_name",
								"name": "SUSE Linux Enterprise Module for Python 3 15 SP6",
								"product": {
									"name": "SUSE Linux Enterprise Module for Python 3 15 SP6",
									"product_id": "SUSE Linux Enterprise Module for Python 3 15 SP6",
									"product_identification_helper": {
										"cpe": "cpe:/o:suse:sle-module-python3:15:sp6"
									}
								}
							},
							{
								"category": "product_name",
								"name": "openSUSE Leap 15.5",
								"product": {
									"name": "openSUSE Leap 15.5",
									"product_id": "openSUSE Leap 15.5",
									"product_identification_helper": {
										"cpe": "cpe:/o:opensuse:leap:15.5"
									}
								}
							},
							{
								"category": "product_name",
								"name": "openSUSE Leap 15.6",
								"product": {
									"name": "openSUSE Leap 15.6",
									"product_id": "openSUSE Leap 15.6",
									"product_identification_helper": {
										"cpe": "cpe:/o:opensuse:leap:15.6"
									}
								}
							}
						],
						"category": "product_family",
						"name": "SUSE Linux Enterprise"
					}
				],
				"category": "vendor",
				"name": "SUSE"
			}
		],
		"relationships": [
			{
				"category": "default_component_of",
				"full_product_name": {
					"name": "python311-urllib3-2.0.7-150400.7.18.1.noarch as component of SUSE Linux Enterprise Module for Public Cloud 15 SP4",
					"product_id": "SUSE Linux Enterprise Module for Public Cloud 15 SP4:python311-urllib3-2.0.7-150400.7.18.1.noarch"
				},
				"product_reference": "python311-urllib3-2.0.7-150400.7.18.1.noarch",
				"relates_to_product_reference": "SUSE Linux Enterprise Module for Public Cloud 15 SP4"
			},
			{
				"category": "default_component_of",
				"full_product_name": {
					"name": "python311-urllib3-2.0.7-150400.7.18.1.noarch as component of SUSE Linux Enterprise Module for Python 3 15 SP5",
					"product_id": "SUSE Linux Enterprise Module for Python 3 15 SP5:python311-urllib3-2.0.7-150400.7.18.1.noarch"
				},
				"product_reference": "python311-urllib3-2.0.7-150400.7.18.1.noarch",
				"relates_to_product_reference": "SUSE Linux Enterprise Module for Python 3 15 SP5"
			},
			{
				"category": "default_component_of",
				"full_product_name": {
					"name": "python311-urllib3-2.0.7-150400.7.18.1.noarch as component of SUSE Linux Enterprise Module for Python 3 15 SP6",
					"product_id": "SUSE Linux Enterprise Module for Python 3 15 SP6:python311-urllib3-2.0.7-150400.7.18.1.noarch"
				},
				"product_reference": "python311-urllib3-2.0.7-150400.7.18.1.noarch",
				"relates_to_product_reference": "SUSE Linux Enterprise Module for Python 3 15 SP6"
			},
			{
				"category": "default_component_of",
				"full_product_name": {
					"name": "python311-urllib3-2.0.7-150400.7.18.1.noarch as component of openSUSE Leap 15.5",
					"product_id": "openSUSE Leap 15.5:python311-urllib3-2.0.7-150400.7.18.1.noarch"
				},
				"product_reference": "python311-urllib3-2.0.7-150400.7.18.1.noarch",
				"relates_to_product_reference": "openSUSE Leap 15.5"
			},
			{
				"category": "default_component_of",
				"full_product_name": {
					"name": "python311-urllib3-2.0.7-150400.7.18.1.noarch as component of openSUSE Leap 15.6",
					"product_id": "openSUSE Leap 15.6:python311-urllib3-2.0.7-150400.7.18.1.noarch"
				},
				"product_reference": "python311-urllib3-2.0.7-150400.7.18.1.noarch",
				"relates_to_product_reference": "openSUSE Leap 15.6"
			}
		]
	},
	"vulnerabilities": [
		{
			"cve": "CVE-2024-37891",
			"ids": [
				{
					"system_name": "SUSE CVE Page",
					"text": "https://www.suse.com/security/cve/CVE-2024-37891"
				}
			],
			"notes": [
				{
					"category": "general",
					"text": " urllib3 is a user-friendly HTTP client library for Python. When using urllib3's proxy support with `ProxyManager`, the `Proxy-Authorization` header is only sent to the configured proxy, as expected. However, when sending HTTP requests *without* using urllib3's proxy support, it's possible to accidentally configure the `Proxy-Authorization` header even though it won't have any effect as the request is not using a forwarding proxy or a tunneling proxy. In those cases, urllib3 doesn't treat the `Proxy-Authorization` HTTP header as one carrying authentication material and thus doesn't strip the header on cross-origin redirects. Because this is a highly unlikely scenario, we believe the severity of this vulnerability is low for almost all users. Out of an abundance of caution urllib3 will automatically strip the `Proxy-Authorization` header during cross-origin redirects to avoid the small chance that users are doing this on accident. Users should use urllib3's proxy support or disable automatic redirects to achieve safe processing of the `Proxy-Authorization` header, but we still decided to strip the header by default in order to further protect users who aren't using the correct approach. We believe the number of usages affected by this advisory is low. It requires all of the following to be true to be exploited: 1. Setting the `Proxy-Authorization` header without using urllib3's built-in proxy support. 2. Not disabling HTTP redirects. 3. Either not using an HTTPS origin server or for the proxy or target origin to redirect to a malicious origin. Users are advised to update to either version 1.26.19 or version 2.2.2. Users unable to upgrade may use the `Proxy-Authorization` header with urllib3's `ProxyManager`, disable HTTP redirects using `redirects=False` when sending requests, or not user the `Proxy-Authorization` header as mitigations.",
					"title": "CVE description"
				}
			],
			"product_status": {
				"recommended": [
					"SUSE Linux Enterprise Module for Public Cloud 15 SP4:python311-urllib3-2.0.7-150400.7.18.1.noarch",
					"SUSE Linux Enterprise Module for Python 3 15 SP5:python311-urllib3-2.0.7-150400.7.18.1.noarch",
					"SUSE Linux Enterprise Module for Python 3 15 SP6:python311-urllib3-2.0.7-150400.7.18.1.noarch",
					"openSUSE Leap 15.5:python311-urllib3-2.0.7-150400.7.18.1.noarch",
					"openSUSE Leap 15.6:python311-urllib3-2.0.7-150400.7.18.1.noarch"
				]
			},
			"references": [
				{
					"category": "external",
					"summary": "CVE-2024-37891",
					"url": "https://www.suse.com/security/cve/CVE-2024-37891"
				},
				{
					"category": "external",
					"summary": "SUSE Bug 1226469 for CVE-2024-37891",
					"url": "https://bugzilla.suse.com/1226469"
				}
			],
			"remediations": [
				{
					"category": "vendor_fix",
					"details": "To install this SUSE Security Update use the SUSE recommended installation methods like YaST online_update or \"zypper patch\".\n",
					"product_ids": [
						"SUSE Linux Enterprise Module for Public Cloud 15 SP4:python311-urllib3-2.0.7-150400.7.18.1.noarch",
						"SUSE Linux Enterprise Module for Python 3 15 SP5:python311-urllib3-2.0.7-150400.7.18.1.noarch",
						"SUSE Linux Enterprise Module for Python 3 15 SP6:python311-urllib3-2.0.7-150400.7.18.1.noarch",
						"openSUSE Leap 15.5:python311-urllib3-2.0.7-150400.7.18.1.noarch",
						"openSUSE Leap 15.6:python311-urllib3-2.0.7-150400.7.18.1.noarch"
					]
				}
			],
			"scores": [
				{
					"cvss_v3": {
						"baseScore": 4.4,
						"baseSeverity": "MEDIUM",
						"vectorString": "CVSS:3.1/AV:N/AC:H/PR:H/UI:N/S:U/C:H/I:N/A:N",
						"version": "3.1"
					},
					"products": [
						"SUSE Linux Enterprise Module for Public Cloud 15 SP4:python311-urllib3-2.0.7-150400.7.18.1.noarch",
						"SUSE Linux Enterprise Module for Python 3 15 SP5:python311-urllib3-2.0.7-150400.7.18.1.noarch",
						"SUSE Linux Enterprise Module for Python 3 15 SP6:python311-urllib3-2.0.7-150400.7.18.1.noarch",
						"openSUSE Leap 15.5:python311-urllib3-2.0.7-150400.7.18.1.noarch",
						"openSUSE Leap 15.6:python311-urllib3-2.0.7-150400.7.18.1.noarch"
					]
				}
			],
			"threats": [
				{
					"category": "impact",
					"date": "2024-07-08T11:14:03Z",
					"details": "moderate"
				}
			],
			"title": "CVE-2024-37891"
		}
	]
}
